### DISCUSSION TITLE
Last Inserted ID for MySQL.

### DISCUSSION THREAD

**Discussion by abhishek-hotstar** [sqlc-dev/sqlc#1206] [Q&A] [answered]
CREATE TABLE IF NOT EXISTS student
(
      id          INT PRIMARY KEY AUTO_INCREMENT,
    name      VARCHAR(50) NOT NULL
);

/* name: InsertStudent :exec */
INSERT INTO student (name)
VALUES (?);

The result only returns an error, not the `sql.Result`. Only on the result pointer can we call `LastInsertId()`. Please suggest a way forward. 

Thank you.

**Comment by aitva** [comment, marked as answer]
You can get an `sql.Result` by using the [`:execresult` command](https://docs.sqlc.dev/en/stable/reference/query-annotations.html#execresult):

```sql
CREATE TABLE IF NOT EXISTS student
(
id INT PRIMARY KEY AUTO_INCREMENT,
name VARCHAR(50) NOT NULL
);

/* name: InsertStudent :execresult */
INSERT INTO student (name)
VALUES (?);
```
